cql3/multi_column_relation.hh

```diff
@@ -99,6 +101,15 @@ public:
         return create_multi_column_relation(std::move(entities), relation_type, std::move(values_or_marker), {}, {});
     }
 
+    /**
+     * Same as above, but sets the magic mode that causes us to treat the restrictions as "raw" clustering bounds
+     */
+    static shared_ptr<multi_column_relation> create_scylla_clustering_bound_non_in_relation(std::vector<shared_ptr<column_identifier::raw>> entities,
+                                                                    expr::oper_t relation_type, shared_ptr<term::multi_column_raw> values_or_marker) {
+        assert(relation_type != expr::oper_t::IN);
+        return create_multi_column_relation(std::move(entities), relation_type, std::move(values_or_marker), {}, {}, mode::scylla_clustering_bound);
```

Comment:
I think we should check that the columns form a prefix of the clustering key, in order. Otherwise (I can't think what would happen otherwise).

Reply:
We do check that, but later:
https://github.com/scylladb/scylla/blob/fb1d3fe2cf41a72190846dde85e3ebd00e6eb4b3/cql3/multi_column_relation.hh#L237

Reply:
Ah. I really dislike how the code says it's a multi-column restriction, but in reality it's a clustering key restriction.

A real multi-column restriction could apply to the partition key (without ordering the tuple, but only with equality relation), or to any column combination (with filtering).

Reply:
That is an idiosyncrasy outside the scope of this change.

Reply:
Of course.